libbeat/common/file/rotator.go

```diff
@@ -243,5 +266,65 @@ func (r *Rotator) rotate() error {
 		}
 	}
 
+	currentFileName := r.backupName(0)
```

Comment:
Could this be moved into a separate method like `compressBackups`. This method is becomong too long.